python/cudf/cudf/_lib/utils.pyx

```diff
@@ -290,6 +289,20 @@ cdef data_from_unique_ptr(
     }
     return data, index
 
+cdef columns_from_unique_ptr(
+    unique_ptr[table] c_tbl
+):
```

Comment:
This function returns a list of columns, whereas the preexisting method returns a dictionary and an index. The long-term plan is to move away from that API entirely and leave the separation of the columns into "data" and "index" to the Python implementation.

Reply:
Even if `data_from_unique_ptr` will eventually be removed, I think it would be better to call this function in `data_from_unique_ptr` rather than re-implement the logic twice. Either way, it's fine with me if this suggestion is declined.

Reply:
Because `_data` is shared across all `Frame` objects, I would assume that `data_from_unique_ptr` to return a dictionary rather than a list of columns. However, I agree we shouldn't implement the same logic twice. We could call `column_from_unique_ptr` from its twins.

Reply:
I think calling one function from the other would be fine to reduce duplication in the short term.